### PR TITLE
Check privileges for symlink tests on Windows

### PR DESCRIPTION
**Pull request by mattn**
Tests for symlink always fail when run the tests as non-admin users on Windows since Windows require privileges to create symlinks. This CL change to check the privileges.

(Can I put label `hacktoberfest-accepted` on this PR ?)

### CODE REVIEW

File: fsnotify_test.go [hunk 1]
```diff
@@ -972,7 +989,8 @@ func TestAdd(t *testing.T) {
 }
 
 // TODO: should also check internal state is correct/cleaned up; e.g. no
-//       left-over file descriptors or whatnot.
+//
+//	left-over file descriptors or whatnot.
```
Comment (mattn) [inline fsnotify_test.go:993]: oop. I'll revert this part and some below.

**Comment by arp242** [comment]
Can you put the `helper_.go` files in the `internal` package? There's already `internal/windows.go` and `internal/unix.go`, although the unix.go one has odd build tags; maybe add a unix2.go.

Just keeps the root a bit cleaner of files 😅

**Comment by mattn** [comment]
Thanks.

**Comment by mattn** [comment]
Ah, newer Go fmt breaks indentation.

**Comment by arp242** [comment]
> Ah, newer Go fmt breaks indentation.

I guess I still use an older `goimports` that I haven't updated since 1.19. Bit annoying it reformats it; I like to indent my TODO comments like that, but can remove the indentation if it's causing gofmt problems.

**Comment by mattn** [comment]
I've had too much coffee and I'm full. ☕😂

image

**Review by arp242 — APPROVED** [review]

**Comment by arp242** [comment]
Yeah, the illumos tests always take forever, although never seen it run that long. I'll just merge it – thanks!

**Comment by mattn** [comment]
Thank you.

**Pull request merged**